File: src/main/java/ui/pageobjects/fragments/modals/LoginModal.java
package ui.pageobjects.fragments.modals;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;

public class LoginModal {
    private final Page page;

    public LoginModal(Page page) {
        this.page = page;
    }

    public Locator title() {
        return this.page.locator("#dialog-title");
    }

    public Locator emailInput() {
        return this.page.locator("#signup_email");
    }

    public Locator passwordInput() {
        return this.page.locator("#login_password");
    }

    public Locator loginButton() {
        return this.page.locator("input[type='submit']");
    }

    public void performLogin(String email, String password) {
        this.emailInput().type(email);
        this.passwordInput().type(password);
        this.loginButton().click();
    }
}

File: src/main/kotlin/ui/pageobjects/fragments/modals/LoginModal.kt
package ui.pageobjects.fragments.modals

import com.microsoft.playwright.Locator
import com.microsoft.playwright.Page

class LoginModal(private val page: Page) {
    fun title(): Locator {
        return page.locator("#dialog-title")
    }

    fun emailInput(): Locator {
        return page.locator("#signup_email")
    }

    fun passwordInput(): Locator {
        return page.locator("#login_password")
    }

    fun loginButton(): Locator {
        return page.locator("input[type='submit']")
    }

    fun performLogin(email: String?, password: String?) {
        emailInput().type(email)
        passwordInput().type(password)
        loginButton().click()
    }
}
